### PR TITLE
feat: Adds dark theme support for public portal [CW-2525]

### PR DESCRIPTION
**Pull request by nithindavid**
## Description
Adds support for public portal through URL params.

**Loom video**
https://www.loom.com/share/8d68da872b694d90b7b7eaa2b061d67c?sid=b8ff4b48-9079-4d11-9a87-55fe36e7b26f

image

image
image

Fixes # (issue)
https://linear.app/chatwoot/issue/CW-2525/support-dark-mode-for-help-center-pages
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by iamsivin — APPROVED** [review]
I tested it locally, and working as expected. Here is the loom video.
https://www.loom.com/share/124031db181e4f9c8836dc327e0b2142?sid=1c2e7351-3c45-4dec-a9c3-8e7f93aeec47

File: app/views/layouts/portal.html.erb [hunk 1]
```diff
@@ -40,6 +40,11 @@ By default, it renders:
       </main>
     </div>
   </body>
+  <style>
+  :root {
+    color-scheme: '<%= @theme %>';
```
Comment (jsobralgitpush) [inline app/views/layouts/portal.html.erb:45]: Grabs the `@theme` using a `<style>` tag is the best practice to solve this?

File: app/views/public/api/v1/portals/categories/show.html.erb [hunk 2]
```diff
@@ -6,23 +6,23 @@
 <% end %>
 
 <% if !@is_plain_layout_enabled %>
-<div class="bg-woot-50 pt-16 pb-20" style="background: url(/assets/images/hc/grid.svg) color-mix(in srgb, <%= @portal.color %> 10%, white)">
+<div class="bg-woot-50 dark:bg-woot-900 pt-16 pb-20" style="background: url(/assets/images/hc/grid.svg) color-mix(in srgb, <%= @portal.color %> 10%, <%= @theme == 'dark' ? 'black' : 'white' %>)">
```
Comment (pranavrajs) [inline app/views/public/api/v1/portals/categories/show.html.erb:9]: `color-mix(in srgb, <%= @portal.color %> 10%, <%= @theme == 'dark' ? 'black' : 'white' %>)`

This portion is repeated everywhere. Can we create a helper method for the same.

File: app/views/layouts/portal.html.erb (hunk 1 above)
Comment (nithindavid) [reply app/views/layouts/portal.html.erb:45]: @jsobralgitpush Thanks for pointing this out. However, we need to set the color-scheme in order to enable dark mode for native browser components such as scrollbars, inputs, select, etc. The value of `@theme` can be either `dark` or `light`. I hope this clarifies the situation. Please let me know if you have a better solution to address this.

File: app/views/public/api/v1/portals/articles/show.html.erb [hunk 3]
```diff
@@ -12,25 +12,25 @@
 <% end %>
 
 <% if !@is_plain_layout_enabled %>
-<div class="bg-woot-50 py-8 shadow-inner" style="background: url(/assets/images/hc/grid.svg) color-mix(in srgb, <%= @portal.color %> 10%, white)">
+<div class="bg-woot-50 dark:bg-woot-900 py-8 shadow-inner" style="background: url(/assets/images/hc/grid.svg) color-mix(in srgb, <%= @portal.color %> 10%, <%= @theme == 'dark' ? 'black' : 'white' %>)">
```
Comment (pranavrajs) [inline app/views/public/api/v1/portals/articles/show.html.erb:15]: Please create a dark variant of the grid and switch it if the color scheme is dark.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Minor comments, please take a look at the individual comments.

File: app/views/public/api/v1/portals/articles/show.html.erb (hunk 3 above)
Comment (nithindavid) [reply app/views/public/api/v1/portals/articles/show.html.erb:15]: 👍

File: app/controllers/public/api/v1/portals/base_controller.rb [hunk 4]
```diff
@@ -9,6 +10,14 @@ def show_plain_layout
     @is_plain_layout_enabled = params[:show_plain_layout] == 'true'
   end
 
+  def set_color_scheme
```
Comment (nithindavid) [inline app/controllers/public/api/v1/portals/base_controller.rb:13]: @sojan-official Please have a look

**Review by sojan-official — APPROVED** [review]

**Pull request merged**